plugins/module_utils/network/ios/facts/vlans/vlans.py

```diff
@@ -43,7 +43,12 @@ def __init__(self, module, subspec="config", options="options"):
         self.generated_spec = utils.generate_dict(facts_argument_spec)
 
     def get_vlans_data(self, connection):
-        return connection.get("show vlan")
+        try:
+            return connection.get("show vlan")
+        except Exception as ex:
+            self._module.fail_json(
+                "'show vlan' doesn't seems to be supported on the respective IOS box, please make sure it's an L2 switch"
```

Comment:
Can add the actual value of exception as well in the error message as well.
For eg: `"'show vlan' command execution failed with error %s" % to_text(ex)`